src/analyses/interval_domain.cpp

```diff
@@ -527,3 +525,53 @@ exprt interval_domaint::make_expression(const symbol_exprt &src) const
   else
     return true_exprt();
 }
+
+/*******************************************************************\
+
+Function: interval_domaint::simplify
+
+  Inputs: The expression to simplify.
+
+ Outputs: A simplified version of the expression.
+
+ Purpose: Uses the abstract state to simplify a given expression
+          using context-specific information.
+
+\*******************************************************************/
+
+bool interval_domaint::ai_simplify(
+  exprt &condition,
+  const namespacet &ns) const
+{
+  bool unchanged=true;
+  interval_domaint d(*this);
+
+  // merge intervals to properly handle conjunction
+  if(condition.id()==ID_and)
+  {
+    interval_domaint a;
+    a.make_top();
+    a.assume(condition, ns);
+
+    if(!a.join(d))
+    {
+      unchanged=condition.is_true();
+      condition.make_true();
+    }
+  }
+  else if(condition.id()==ID_symbol)
+  {
+    // TODO: we have to handle symbol expression
+  }
+  else
+  {
+    d.assume(not_exprt(condition), ns);
```

Comment:
That *is* the disjunction case :-)  There are three things going on here:

1. We want to implement this primarily to be able to check assertions, so reducing to true is sufficient.  We can do better and in the new interval domain we are writing, we will, so there is a limit to how much code we want to write here.

2. We could work through all of the cases of exprt here but that code has to be in assume as well to handle branching.  So rather than have a wonky duplication, we really want to implement this using assume, merge, top, bottom, etc.  which is sufficient to reduce to true.

3. But converting the expression to a domain and doing domain things means we are limited to what the domain can express.  Thus disjuncts are a problem.  1 <= a && a <= 2  can be repesented exactly but  a < 1 || a > 2 can't.  The logic for "put this expression into the most representable normal form" should be in assume a it will be needed there anyway.  Thus this call, with the negation, is meant to help it in the case of disjunctions.

How much of that should have been in the comments? :-)